src/main/java/com/bookstore/bookstoreapi/data/controller/BookController.java

```diff
@@ -23,4 +25,15 @@ public ResponseEntity<Book> createABook(@RequestBody Book book) {
     }
     return new ResponseEntity<>(createdBook.get(), HttpStatus.CREATED);
   }
+
+  @GetMapping("/books/findbyisbn")
+  public ResponseEntity<Book> getBookByISBN(@RequestParam String isbn) {
+    Book book = bookService.getBookByISBN(isbn);
+
+    if (book != null) {
+      return new ResponseEntity<>(book, HttpStatus.OK);
+    } else {
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
No need for else statement here, because return statement is an exit statement, if `if` statement is executed, it will not go any further, if I overlooked this elsewhere, please correct it.